dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -310,6 +313,48 @@ RSOperatorPtr parseTiExpr(const tipb::Expr & expr,
             op = parseTiCompareExpr(expr, filter_type, columns_to_read, creator, timezone_info, log);
             break;
 
+        case FilterParser::RSFilterType::IsNull:
+        {
+            // for IsNULL filter, we only support do filter when the child is ColumnExpr.
+            // That is, we only do filter when the statement likes `where a is null`, but not `where (a > 1) is null`
+            // because in other filter calculation(like Equal/Less/LessEqual/Greater/GreateEqual/NotEqual), we just make filter ignoring the null value.
+            // Therefore, if we support IsNull with sub expr, there could be correctness problem.
+            // For example, we have a table t(a int, b int), and the data is: (1, 2), (0, null), (null, 1)
+            // and then we execute `select * from t where (a > 1) is null`, we want to get (null, 1)
+            // but in RSResult (a > 1), we will get the result RSResult::None, and then we think the result is the empty set.
+            if (unlikely(expr.children_size() != 1))
+            {
+                op = createUnsupported(
+                    expr.ShortDebugString(),
+                    "filter IsNull with " + DB::toString(expr.children_size()) + " children",
+                    false);
+            }
+            else
+            {
+                const auto & child = expr.children(0);
+                if (likely(isColumnExpr(child)))
```

Comment:
Maybe we can't assume that users don't write sql like `isnull(function(xx))`

Reply:
Do you means the statement like where isnull(a>1)?
But now we only assume the filter IsNull can handle the case for column only, just like isnull(a), but not isnull(a>1) or a > 1 is null, because it may cause some correctness problem. So for the statement like where isnull(a>1), I think we should make the filter unsupported.